src/workerd/api/headers.c++

```diff
@@ -53,158 +154,279 @@ void warnIfBadHeaderString(const jsg::ByteString& byteString) {
   }
 }
 
+constexpr bool isHttpWhitespace(char c) {
+  return c == '\t' || c == '\r' || c == '\n' || c == ' ';
```

Comment:
We tend to have this in multiple places, we should create a lookup table and reuse it. This is extremely slow.